File: iosApp/iosApp/Views/Components/PostCard.swift
import SwiftUI

struct PostCard: View {
    let userImage: String
    let userName: String
    let timeAgo: String
    let postImage: String
    let postContent: String
    let rating: Int
    let showEditButton: Bool
    let onEditClick: () -> Void

    init(
        userImage: String = "",
        userName: String,
        timeAgo: String,
        postImage: String = "",
        postContent: String,
        rating: Int = 5,
        showEditButton: Bool = false,
        onEditClick: @escaping () -> Void = {}
    ) {
        self.userImage = userImage
        self.userName = userName
        self.timeAgo = timeAgo
        self.postImage = postImage
        self.postContent = postContent
        self.rating = rating
        self.showEditButton = showEditButton
        self.onEditClick = onEditClick
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // User info header
            HStack {
                // User profile image
                Image("winemap_logo") // Replace with actual user image
                    .resizable()
                    .frame(width: 40, height: 40)
                    .clipShape(Circle())

                VStack(alignment: .leading, spacing: 2) {
                    Text(userName)
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.black)

                    Text(timeAgo)
                        .font(.system(size: 12))
                        .foregroundColor(.gray)
                }

                Spacer()

                // Edit button (only for user's own posts)
                if showEditButton {
                    Button(action: onEditClick) {
                        Image(systemName: "pencil")
                            .font(.system(size: 16))
                            .foregroundColor(.gray)
                            .frame(width: 32, height: 32)
                    }
                }
            }

            // Post image
            Image("winemap_logo") // Replace with actual post image
                .resizable()
                .frame(height: 200)
                .cornerRadius(12)
                .clipped()

            // Post content
            Text(postContent)
                .font(.system(size: 14))
                .foregroundColor(.black)
                .lineSpacing(6)

            // Star rating
            StarRating(rating: rating, size: 16)
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(16)
        .shadow(radius: 4)
        .padding(.vertical, 8)
    }
}

File: androidApp/src/main/java/com/example/winemap/android/ui/components/PostCard.kt
package com.example.winemap.android.ui.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun PostCard(
    userImage: String = "",
    userName: String,
    timeAgo: String,
    postImage: String = "",
    postContent: String,
    rating: Int = 5,
    onEditClick: () -> Unit = {},
    showEditButton: Boolean = false
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            // User info header
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                // User profile image
                Image(
                    painter = painterResource(id = com.example.winemap.android.R.drawable.winemap_logo),
                    contentDescription = "User Profile",
                    modifier = Modifier
                        .size(40.dp)
                        .clip(CircleShape),
                    contentScale = ContentScale.Crop
                )

                Spacer(modifier = Modifier.width(12.dp))

                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = userName,
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.Black
                    )
                    Text(
                        text = timeAgo,
                        fontSize = 12.sp,
                        color = Color.Gray
                    )
                }

                // Edit button (only for user's own posts)
                if (showEditButton) {
                    IconButton(
                        onClick = onEditClick,
                        modifier = Modifier.size(32.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Edit,
                            contentDescription = "Edit Post",
                            tint = Color.Gray,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // Post image
            Image(
                painter = painterResource(id = com.example.winemap.android.R.drawable.winemap_logo),
                contentDescription = "Post Image",
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp)
                    .clip(RoundedCornerShape(12.dp)),
                contentScale = ContentScale.Crop
            )

            Spacer(modifier = Modifier.height(12.dp))

            // Post content
            Text(
                text = postContent,
                fontSize = 14.sp,
                color = Color.Black,
                lineHeight = 20.sp
            )

            Spacer(modifier = Modifier.height(8.dp))

            // Star rating
            StarRating(
                rating = rating,
                size = 16.dp
            )
        }
    }
}
